### PR TITLE
[REF] stock: convert client action "stock_report_generic"

### PR DESCRIPTION
**Pull request by FrancoisGe**
This commit is to convert the "stock_report_generic" client action into owl.

TaskID: 3175084

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                    model_name: line.model,
+                    level: line.level + 30 || 1,
+                })
+            ).map(processLine);
+        }
+    }
+}
+
+TraceabilityReport.template = "stock.TraceabilityReport";
+TraceabilityReport.components = { Layout };
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.js:149]: static

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    <t t-name="stock.TraceabilityReport" owl="1">
+        <Layout display="display">
+            <t t-set-slot="layout-buttons">
+                <div class="o_cp_buttons" role="toolbar" aria-label="Control panel buttons" t-ref="buttons">
+                    <button type="button" class='btn btn-primary o_stock-widget-pdf' t-on-click="() => this.onClickPrint()">PRINT</button>
+                </div>
+            </t>
+
+
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.xml:12]: remove

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <t t-name="stock.TraceabilityReport" owl="1">
+        <Layout display="display">
+            <t t-set-slot="layout-buttons">
+                <div class="o_cp_buttons" role="toolbar" aria-label="Control panel buttons" t-ref="buttons">
+                    <button type="button" class='btn btn-primary o_stock-widget-pdf' t-on-click="() => this.onClickPrint()">PRINT</button>
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.xml:8]: Print

Comment (aab-odoo) [reply addons/stock/static/src/client_actions/stock_traceability_report_backend.xml:8]: remove class o_stock-widget-pdf

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                </div>
+            </t>
+
+
+            <div class="container-fluid o_stock_reports_page o_stock_reports_no_print">
+                <span t-if="state.lines.length" class="o_report_heading text-start"><h1>Traceability Report</h1></span>
+                <span><t t-esc="context"/></span>
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.xml:15]: empty span?

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                            </thead>
+                            <tbody>
+                                <t t-call="stock.ReportMRPLines">
+                                    <t t-set="lines" t-value="state.lines"/>
+                                    <t t-set="hasUpDown" t-value="true"/>
+                                </t>
+                            </tbody>
+                        </table>
+                    </t>
+                    <span t-else="" class="text-center"><h1>No operation made on this lot.</h1></span>
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.xml:38]:
```xml
<t t-if="lines.length>
 <h1/>
 content
</t>
<t t-else=""/>
  <other h1/>
</t>
```

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            data.push(...extractPrintData(line.lines));
+        }
+    }
+    return data;
+}
+
+export class TraceabilityReport extends Component {
+    setup() {
+        this.actionService = useService("action");
+        this.rpc = useService("rpc");
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.js:35]: remove

**Review by aab-odoo — APPROVED** [review]
Hello @svs-odoo 

May I ask you to review this PR? Thanks.

File: addons/stock/static/src/client_actions/stock_traceability_report_backend.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+        this.state = useState({
+            lines: (this.props.state && this.props.state.lines) || [],
+        });
+
+        const { active_id, active_model, auto_unfold, context, lot_name, ttype, url } =
+            this.props.action.context;
+        this.controllerUrl = url;
+
+        this.givenContext = context
```
Comment (aab-odoo) [inline addons/stock/static/src/client_actions/stock_traceability_report_backend.js:54]: this.context ?

**Review by svs-odoo — COMMENTED** [review]
LGTM, the only issue I found is there is no more scrollbar when the report is too long.
I also left some minor suggestions

**Comment by svs-odoo** [comment]
robodoo r+

**Pull request closed without merging**